pkg/rpcclient/nep17/nep17.go

```diff
@@ -0,0 +1,141 @@
+func NewReader(invoker Invoker, hash util.Uint160) *TokenReader {
+	return &TokenReader{*neptoken.New(invoker, hash), invoker, hash}
+}
+
+// New creates an instance of Token for contract with the given hash
+// using the given Actor.
+func New(actor Actor, hash util.Uint160) *Token {
+	return &Token{*NewReader(actor, hash), actor}
+}
+
+// BalanceOf returns the token balance of the given account.
+func (t *TokenReader) BalanceOf(account util.Uint160) (*big.Int, error) {
+	return unwrap.BigInt(t.invoker.Call(t.hash, "balanceOf", account))
+}
+
+// Transfer creates and sends a transaction that performs a `transfer` method
+// call using the given parameters and checks for this call result, failing the
+// transaction if it's not true. The returned values are transaction hash, its
+// ValidUntilBlock value and an error if any.
+func (t *Token) Transfer(from util.Uint160, to util.Uint160, amount *big.Int, data interface{}) (util.Uint256, uint32, error) {
+	script, err := t.transferScript(from, to, amount, data)
+	if err != nil {
+		return util.Uint256{}, 0, err
+	}
+	return t.actor.SendRun(script)
+}
+
+// TransferTransaction creates a transaction that performs a `transfer` method
+// call using the given parameters and checks for this call result, failing the
+// transaction if it's not true. This transaction is signed, but not sent to the
+// network, instead it's returned to the caller.
+func (t *Token) TransferTransaction(from util.Uint160, to util.Uint160, amount *big.Int, data interface{}) (*transaction.Transaction, error) {
+	script, err := t.transferScript(from, to, amount, data)
+	if err != nil {
+		return nil, err
+	}
+	return t.actor.MakeRun(script)
+}
+
+// TransferUnsigned creates a transaction that performs a `transfer` method
+// call using the given parameters and checks for this call result, failing the
+// transaction if it's not true. This transaction is not signed and just returned
+// to the caller.
+func (t *Token) TransferUnsigned(from util.Uint160, to util.Uint160, amount *big.Int, data interface{}) (*transaction.Transaction, error) {
+	script, err := t.transferScript(from, to, amount, data)
+	if err != nil {
+		return nil, err
+	}
+	return t.actor.MakeUnsignedRun(script, nil)
+}
+
+func (t *Token) transferScript(from util.Uint160, to util.Uint160, amount *big.Int, data interface{}) ([]byte, error) {
+	return smartcontract.CreateCallWithAssertScript(t.hash, "transfer", from, to, amount, data)
+}
+
+// MultiTransfer is not a real NEP-17 method, but rather a convenient way to
+// perform multiple transfers (usually from a single account) in one transaction.
+// It accepts a set of parameters, creates a script that calls `transfer` as
+// many times as needed (with ASSERTs added, so if any of the transfers fail
+// whole transaction (with all transfers) fails). The values returned are the
+// same as in Transfer.
+func (t *Token) MultiTransfer([]TransferParameters) (util.Uint256, uint32, error) {
+	return util.Uint256{}, 0, nil
```

Comment:
Maybe remove it for now, and add the whole method later?

Reply:
It wasn't here before, is it added by accident?

Reply:
Ouch. Looks like I've committed something that wasn't supposed to be committed at the moment.

Reply:
Fixed.